lib/runner/matchers/tags.js

```diff
@@ -2,25 +2,28 @@ const Logger = require('../../util/logger.js');
 
 class TagsMatcher {
   constructor(settings) {
-    this.settings = settings;
+    this.settings = {
```

Comment:
I see. Still, I think the solution should be in that case to update the tests that are problematic, since this approach can create confusion and inconsistencies later on.

Reply:
I disagree, because that would make unit tests unnecessary verbose.

Reply:
I have encapsulated tag parsing logic in tags.js. I think it looks better now.

Reply:
> I disagree because that would make unit tests unnecessary verbose.

Even so, having verbose tests doesn't create confusion. The `settings` or `this.settings`, wherever it appears throughout the codebase should be consistent. So in this case, to avoid changing the tests unnecessarily, we could create two different properties `this.tagFilter` and `this.skiptagsFilter` instead maybe?

Reply:
Thats a good idea!  I went for `this.includeTags` and `this.excludeTags`.
I also applied the other changes you suggested.